src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal.OpenSsl
+{
+    internal sealed partial class PkcsPalOpenSsl : PkcsPal
+    {
+        internal PkcsPalOpenSsl()
+        {
+        }
+
+        public sealed override byte[] Encrypt(
+            CmsRecipientCollection recipients,
+            ContentInfo contentInfo,
+            AlgorithmIdentifier contentEncryptionAlgorithm,
+            X509Certificate2Collection originatorCerts,
+            CryptographicAttributeObjectCollection unprotectedAttributes)
+        {
+            // TODO(3334): see CMS_ContentInfo *CMS_encrypt(...) in cms.h
+            throw new NotImplementedException();
+        }
+
+        public sealed override DecryptorPal Decode(byte[] encodedMessage,
+            out int version,
+            out ContentInfo contentInfo,
+            out AlgorithmIdentifier contentEncryptionAlgorithm,
+            out X509Certificate2Collection originatorCerts,
+            out CryptographicAttributeObjectCollection unprotectedAttributes)
+        {
+            return DecryptorPalOpenSsl.Decode(
+                encodedMessage,
+                out version,
+                out contentInfo,
+                out contentEncryptionAlgorithm,
+                out originatorCerts,
+                out unprotectedAttributes);
+        }
+
+        public sealed override byte[] EncodeOctetString(byte[] octets)
+        {
+            return DerEncoder.EncodeOctetString(octets);
+        }
+
+        public sealed override byte[] DecodeOctetString(byte[] encodedOctets)
+        {
+            try
+            {
+                return DerSequenceReader.CreateForPayload(encodedOctets).ReadOctetString();
+            }
+            catch (Exception e) when (e is InvalidOperationException || e is IndexOutOfRangeException)
```

Comment:
What's generating an IndexOutOfRangeException?  That typically happens when one indexes out of range of an array, and ideally we'd be doing checks before accessing the array to avoid such invalid accesses rather than subsequently catching the exception.